sdk/identity/identity/src/credentials/interactiveBrowserCredential.browser.ts

```diff
@@ -55,46 +80,123 @@ export class InteractiveBrowserCredential implements TokenCredential {
 
+    this.correlationId = options.correlationId;
+
     this.msalConfig = {
       auth: {
         clientId: options.clientId!, // we just initialized it above
         authority: `${options.authorityHost}/${options.tenantId}`,
-        knownAuthorities,
-        ...(options.redirectUri && { redirectUri: options.redirectUri }),
-        ...(options.postLogoutRedirectUri && { redirectUri: options.postLogoutRedirectUri })
+        knownAuthorities
       },
       cache: {
-        cacheLocation: "localStorage",
-        storeAuthStateInCookie: true
+        cacheLocation: "sessionStorage",
+        storeAuthStateInCookie: true // Set to true to improve the experience on IE11 and Edge.
+      },
+      system: {
+        loggerOptions: {
+          loggerCallback: (level, message, containsPii) => {
+            if (containsPii) {
+              return;
+            }
+            switch (level) {
+              case msalBrowser.LogLevel.Error:
+                logger.info(`MSAL Browser V2 error: ${message}`);
+                return;
+              case msalBrowser.LogLevel.Info:
+                logger.info(`MSAL Browser V2 info message: ${message}`);
+                return;
+              case msalBrowser.LogLevel.Verbose:
+                logger.info(`MSAL Browser V2 verbose message: ${message}`);
+                return;
+              case msalBrowser.LogLevel.Warning:
+                logger.info(`MSAL Browser V2 warning: ${message}`);
+                return;
+            }
+          }
+        }
       }
     };
 
-    this.msalObject = new msal.UserAgentApplication(this.msalConfig);
+    this.msalConfig.auth.redirectUri =
+      typeof options.redirectUri === "function" ? options.redirectUri() : options.redirectUri;
+
+    this.msalConfig.auth.postLogoutRedirectUri =
+      typeof options.postLogoutRedirectUri === "function"
+        ? options.postLogoutRedirectUri()
+        : options.postLogoutRedirectUri;
+
+    this.msalObject = new msalBrowser.PublicClientApplication(this.msalConfig);
+
+    this.handleRedirectResponse();
+  }
+
+  /**
+   * Handles the redirect response based on the credentials available at page load.
+   */
+  private async handleRedirectResponse(): Promise<void> {
+    try {
+      const result = await cachedRedirectResponse(this.msalObject);
+      if (result && result.account) {
+        logger.info(`MSAL Browser V2 redirect authentication successful.`);
+        this.account = result.account;
+      } else {
+        this.loadCachedAccount();
+      }
+    } catch (e) {
+      logger.info(`Failed to acquire token through the MSAL redirect login method. ${e.message}`);
+    }
   }
 
-  private login(): Promise<msal.AuthResponse> {
+  /**
+   * Loads the specific target account from the MSAL cache, if it exists.
+   * Returns true if the load was successful.
+   */
+  private loadCachedAccount(): boolean {
+    const accounts = this.msalObject.getAllAccounts();
```

Comment:
TODO: Instead of getting all accounts, always use the cached this.account to get an account by some known ID. this.account should already be populated if either popup or redirect finished athenticating.